### DISCUSSION TITLE
Multiple query parameter empty arrays append orphaned ampersands to query string

### DISCUSSION THREAD

**Discussion by BlakeSzabo** [openapi-ts/openapi-typescript#1934] [Q&A]
Hi,

I've been using `openapi-fetch` and have been enjoying it so far but I have noticed a strange behaviour with arrays being passed in query params

here's a basic implementation of a client that will call the `/posts` endpoint and pass in four separate query params, each an empty array just to be a bit dramatic:
```ts
import createClient from 'openapi-fetch';
import type { paths } from '../types/api';

const client = createClient<paths>({ baseUrl: 'http://localhost:3001' });

export const getPosts = () => {
    client.GET('/posts', {
        params: {
            query: {
                first_array: [],
                second_array: [],
                third_array: [],
                fourth_array: [],
            },
        },
    });
};
```

Given all these arrays are empty, I wouldn't expect them to impact the query params; similar to how `undefined` or `null` would behave. However, the resulting request URL is `http://localhost:3001/posts?&&&`
![image](https://github.com/user-attachments/assets/055b9903-1d74-4db2-8fc5-8a46d0039011)

I assume this behaviour isn't intended but wanted to check before raising an issue. I'm also willing to contribute to help solve this if I can. Thanks.

**Comment by drwpow** [comment]
Thanks for filing! This is an issue for sure. Migrated here, and would love a PR if anyone is able: https://github.com/openapi-ts/openapi-typescript/issues/1935

Should be a simple matter of adding a test first, then getting that test to pass.
